FirebaseMLModelDownloader/Sources/ModelDownloader.swift

```diff
@@ -49,7 +50,28 @@ public enum ModelDownloadType {
 }
 
 /// Downloader to manage custom model downloads.
-public struct ModelDownloader {
+public class ModelDownloader {
+  /// FirebaseApp associated with this instance of ModelDownloader.
+  private let app: FirebaseApp
+
+  private init(app: FirebaseApp) {
+    self.app = app
```

Comment:
Instead of storing the app, it may make sense to store the properties that we need out of the app. Unless it's quite a few, but would make testing easier likely.

Reply:
During initialization also, since we're not integrating with components yet, can you add a notification listener for the string `FIRAppDeleteNotification`? It gets posted here: https://github.com/firebase/firebase-ios-sdk/blob/d098e5a8b0750254d49fd24758f5931e161c4ef8/FirebaseCore/Sources/FIRApp.m#L305

`userInfo` has a key `FIRAppNameKey` that contains the name of the app being deleted. Then, check the static instances (or the current instance) if that name matches and clear the local instances of that app name. 

Since the symbols for the notification name and key are internal constants, using the strings themselves is fine for now since it's a temporary measure and they haven't changed in years.

Reply:
I added a notification observer for app deletion and a shared dictionary to manage instances. 

But I'm not sure how to clean up app instances... Right now when an app is deleted, I'm only removing the instance from the dictionary. What else should I be doing?

Reply:
Whatever local state you think is necessary. Potentially deleting the local models for that name, since it could be configured with different FIROptions on next run.

Reply:
@ryanwilson tagging in case you missed this in the flurry of comments :)